lib/sequence.js

```diff
@@ -167,6 +167,40 @@ Sequence.prototype._setHooks = function() {
             done();
         }.bind(doc));
     });
+
+    this._schema.pre('findOneAndUpdate', function(next) {
+        // jscs:disable
+        var _thisquery = this,
+            referenceValue;
+
+        // jscs:enable
+
+        this._update['$setOnInsert'] = this._update['$setOnInsert'] || {};
+
+        referenceValue = _this._getCounterReferenceField(this._update['$setOnInsert']);
+        _this._setNextCounter(_this._options.reference_fields, referenceValue, function(err, seq) {
+            if (err) return next();
+            _thisquery._update['$setOnInsert'][_this._options.inc_field] = seq;
+            next();
+        }.bind(this._update['$setOnInsert']));
```

Comment:
Honestly, I'm not really sure what the bind was for so I just left it in.
If it's not needed, which it seems it's not, then I'll just remove it.